extensions/zendesk/v1/actions/createTask/config/fields.ts

```diff
@@ -0,0 +1,83 @@
+import { z, type ZodTypeAny } from 'zod'
+import {
+  type Field,
+  FieldType,
+  StringType,
+  DateTimeOptionalSchema,
+  NumericIdSchema,
+  makeStringOptional,
+} from '@awell-health/extensions-core'
+import { ResourceType } from '../../../client/types'
+
+export const fields = {
+  content: {
+    label: 'Content',
+    id: 'content',
+    type: FieldType.STRING,
+    stringType: StringType.TEXT,
+    required: true,
+    description: 'Content of the task.',
+  },
+  dueDate: {
+    label: 'Due date',
+    id: 'dueDate',
+    // Time will be auto-set to 00:00:00 as FieldType.DATE provides only date
+    type: FieldType.DATE,
+    required: false,
+    description: "Date of task's due date in UTC (ISO8601 format).",
+  },
+  ownerId: {
+    label: 'Owner ID',
+    id: 'ownerId',
+    type: FieldType.NUMERIC,
+    required: false,
+    description:
+      'Unique identifier of the user the task is assigned to. Defaults to the unique identifier of the user who created the task.',
+  },
+  resourceType: {
+    label: 'Resource type',
+    id: 'resourceType',
+    type: FieldType.STRING,
+    required: false,
+    description:
+      'Name of the resource type the task is attached to. Possible values: "lead", "contact", "deal".',
+  },
+  resourceId: {
+    label: 'Resource ID',
+    id: 'resourceId',
+    type: FieldType.NUMERIC,
+    required: false,
+    description: 'Unique identifier of the resource the task is attached to.',
+  },
+  completed: {
+    label: 'Completed?',
+    id: 'completed',
+    type: FieldType.BOOLEAN,
+    required: false,
+    description: 'Indicator of whether the task is completed or not.',
+  },
+  remindAt: {
+    label: 'Remind at',
+    id: 'remindAt',
+    // Time will be auto-set to 00:00:00 as FieldType.DATE provides only date
+    type: FieldType.DATE,
+    required: false,
+    description:
+      'Date that we should send you a reminder about the task in UTC (ISO8601 format).',
+  },
+} satisfies Record<string, Field>
+
+const resourceTypeEnum = z.enum<
+  ResourceType,
+  [ResourceType, ...ResourceType[]]
+>([ResourceType.CONTACT, ResourceType.DEAL, ResourceType.LEAD])
+
+export const FieldsValidationSchema = z.object({
+  content: z.string().nonempty(),
+  dueDate: DateTimeOptionalSchema,
+  ownerId: makeStringOptional(NumericIdSchema),
```

Comment:
Just double-checking whether this is correct: `makeStringOptional(NumericIdSchema)`

ownerId is a number, as indicated by `NumericIdSchema` but then you wrap it in `makeStringOptional` which explicitly says "make this string optional" (while it's a number).

Reply:
Yes, I think `makeStringOptional` should be named a little different, but it seems that's the best I could come up at the time I was making it. Under the hood this utility function is `z.union([schema.optional(), z.literal('').transform(() => undefined)])`, which means that you can pass any schema there and it just marks a schema as optional and makes sure that if empty string `''` is passed it is still `undefined` (e.g. situation when you set sth in Awell, but then clear the input)